### PR TITLE
Populate VisualNodes statically instead of dynamically

### PR DESCRIPTION
**Pull request by bollu**
This helps improve auto-completion in IDEs that access static information from the module.
Both Sublime Text and PyCharm benefit from this change. I believe Valloric's `YouCompleteMe` will also benefit from this, as will emacs' `Jedi`, though I haven't tested either (on Windows right now)

One thing is that the docs don't show up (since the docs are dynamically appended to the elements in `create_visual_node`. However, `help(vispy.scene.FooVisual)` works as expected. I was wondering if I should add a one-liner above each declaration.

I also cleaned up the whitespace in the file.

### CODE REVIEW

**Comment by bollu** [comment]
@Eric89GXL: ping

File: vispy/scene/visuals.py [hunk 1]
```diff
@@ -198,25 +198,47 @@ def generate_docstring(subclass, clsname):
...
-    obj = getattr(visuals, obj_name)
-    if (isinstance(obj, type) and 
-       issubclass(obj, visuals.BaseVisual) and 
-       obj is not visuals.Visual):
-        cls = create_visual_node(obj)
-        globals()[cls.__name__] = cls
-        __all__.append(cls.__name__)
+for (obj_name, obj) in globals().items():
+    if (isinstance(obj, type) and issubclass(obj, VisualNode)):
+        __all__.append(obj_name)
```
Comment (larsoner) [inline vispy/scene/visuals.py]: we could just make this a sweet list comprehension

**Comment by larsoner** [comment]
Please add a comment that we need to do this in order for autocomplete to work in IDEs (and probably IPython, too), otherwise LGTM

**Comment by bollu** [comment]
Yep, done as requested.

**Comment by larsoner** [comment]
LGTM, +1 for merge

**Comment by larsoner** [comment]
@bollu can you close this and open it against the `vispy/scenegraph-update` branch? There is a merge conflict, so you might just want to manually copy the relevant changes in `scene/visual.py`.

**Pull request closed without merging**